DependencyInjection/Compiler/TokenStorageCompilerPass.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the FOSOAuthServerBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\OAuthServerBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Andras Ratz <ratz.andras86@gmail.com>
+ */
+class TokenStorageCompilerPass implements CompilerPassInterface
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $definition = $container->getDefinition('fos_oauth_server.security.authentication.listener');
+
+        if ($container->hasDefinition('security.token_storage')) {
+            $tokenStorageReference = new Reference('security.token_storage');
```

Comment:
I'm not sure I understand what you want me to do here however? token_storage is not available on symfony 2.3, and it will throw an exception if left on the xml file; likewise security.context is only available from symfony 2.6. Unless I have the wrong end of the stick here?